server/server.go

```diff
@@ -187,8 +187,12 @@ type Server struct {
 	hotRegionStorage *storage.HotRegionStorage
 	// Store as map[string]*grpc.ClientConn
 	clientConns sync.Map
-	// Store as map[string]*tsopb.TSOClient
-	tsoClients sync.Map
+
+	tsoClientPool struct {
+		mux     sync.RWMutex
+		clients map[string]tsopb.TSO_TsoClient
+	}
```

Comment:
BTW, this could help reduce the code:

```suggestion
	tsoClientPool struct {
		sync.RWMutex
		clients map[string]tsopb.TSO_TsoClient
	}
```

Then:

```go
s.tsoClientPool.Lock()
```

Reply:
changed